<?xml version="1.0"?>
<launch>
  <arg name='host' default='localhost'/>
  <arg name='port' default='2000'/>
  <arg name="role_name" default="ego_vehicle"/>
  <!-- town : "Town01", "Town02", "Town03", "Town04", "Town05", "random"-->
  <arg name="town" default="Town03"/>
  <arg name="width" default="1920"/>
  <arg name="height" default="1080"/>

  <group ns="g29">
    <node pkg="g29_carla" type="manual_control" name="manual_control" required="true" output="screen">
      <param name="img_dir" value="$(find g29_carla)/img"/>
      <param name="role_name" value="$(arg role_name)"/>
      <param name="town" value="$(arg town)"/>
      <param name="sensor_config_path" value="$(find g29_carla)/config/g3od.json"/>
      <param name="vehicle_filter" value="vehicle.tesla.model3"/>
      <param name="force_min" value="0.21"/>
      <param name="host" value="$(arg host)"/>
      <param name="port" value="$(arg port)"/>
      <param name="width" value="$(arg width)"/>
      <param name="height" value="$(arg height)"/>
      <!--
        <param name="hz" value="20"/>
        <param name="kp" value="1.0"/>
        <param name="ki" value="0.0"/>
        <param name="kd" value="0.0"/>
        <param name="force_max" value="1.0"/>
        <param name="event_path" value="/dev/input/event-g29"/>
      -->
    </node>
  </group>

</launch>